hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/DynamoDBMetadataStore.java

```diff
@@ -486,10 +486,12 @@ public void forgetMetadata(Path path) throws IOException {
    * There is no check as to whether the entry exists in the table first.
    * @param path path to delete
    * @param tombstone flag to create a tombstone marker
+   * @param ttlTimeProvider
    * @throws IOException I/O error.
    */
   @Retries.RetryTranslated
-  private void innerDelete(final Path path, boolean tombstone)
+  private void innerDelete(final Path path, boolean tombstone,
```

Comment:
do you think we should check for an entry here at all? I was wondering if it is superflious, but now I think "we really do need it"

Reply:
What do you mean by `check for an entry`? 

> There is no check as to whether the entry exists in the table first.

Reply:
I was wondering if we needed a call to checkPath, but the answer is: yes